packages/xorg_libs.rb

```diff
@@ -0,0 +1,272 @@
+      else
+        puts "Downloading #{filename}".yellow
+        system('curl', '-s', '-C', '-', '--insecure', '-L', '-#', url, '-o', filename)
+        abort 'Checksum mismatch. :/ Try again.'.lightred unless
+          Digest::SHA256.hexdigest( File.read("./#{filename}") ) == sha256sum
+        puts "#{filename} downloaded".lightgreen
+      end
+      
+      system "tar", "xf", "#{filename}", "-C", "xorg_lib"
+      puts "#{filename} unpacked".lightgreen
+    end
+    
+    ############################################################
+    puts "All xorg library source code are unpacked!".lightgreen
+    ############################################################
+    
+  end
+  
+  
+  def self.build
+    Dir.chdir("xorg_lib") do
+    ############################################################
+    puts "Start compiling xorg library from source".lightblue
+    ############################################################
+      for index in 0 ... $xorg_lib_source_name.size
+        folder_name = "#{$xorg_lib_source_name[index]}"
+        folder_name.slice! ".tar.gz"  # remove .tar.gz
+  
+        puts "Configuring #{folder_name} ...".lightblue
+        Dir.chdir(folder_name) do
+         
+          if folder_name.include? "libICE"
+            puts "libICE".lightred
+            system "./configure",
+                   "--prefix=#{CREW_PREFIX}",
+                   "--libdir=#{CREW_LIB_PREFIX}",
+                   "ICE_LIBS=-lpthread"     # This fixes a change in glibc which breaks pulseaudio at runt time
+          end
+          
+          if folder_name.include? "libXfont2"
+            puts "libXfont2".lightred
+            system "./configure",
+                   "--prefix=#{CREW_PREFIX}",
+                   "--libdir=#{CREW_LIB_PREFIX}",
+                   "--disable-static",
+                   "--enable-shared",
+                   "--disable-devel-docs"
+          end
+          
+          if folder_name.include? "libX11"
+            puts "libXfont2".lightred
+            system "./configure",
+                   "--prefix=#{CREW_PREFIX}",
+                   "--libdir=#{CREW_LIB_PREFIX}",
+                   "--disable-static",
+                   "--enable-shared",
+                   "--enable-loadable-i18n"
+          end
+            
+          if folder_name.include? "libXt-"
+            puts "libXt-1".lightred
+            system "./configure",
+                   "--prefix=#{CREW_PREFIX}",
+                   "--libdir=#{CREW_LIB_PREFIX}",
+                   "--disable-static",
+                   "--enable-shared",
+                   "--with-appdefaultdir=#{CREW_PREFIX}/etc/X11/app-defaults"
+          else
+            puts "#{folder_name}".lightred
+            system "./configure",
+                   "--prefix=#{CREW_PREFIX}",
+                   "--libdir=#{CREW_LIB_PREFIX}",
+                   "--disable-static",
+                   "--enable-shared"
```

Comment:
I don't think you are going to want this else condition.  This will execute every time in the loop when folder_name does not include libXt-.  Use case...when statements instead?